src/lib/Core.hs

```diff
@@ -287,6 +287,18 @@ instance (Color c, SinkableE ann, ToBinding ann c) => BindsEnv (BinderP c ann) w
   toEnvFrag (b:>ann) = EnvFrag (RecSubstFrag (b @> toBinding ann')) Nothing
     where ann' = withExtEvidence b $ sink ann
 
+instance (SinkableE ann, ToBinding ann AtomNameC) => BindsEnv (NonDepNest ann) where
+  toEnvFrag (NonDepNest topBs topAnns) = toEnvFrag $ zipNest topBs topAnns
+    where
+      zipNest :: Distinct l => Nest AtomNameBinder n l -> [ann n] -> Nest (BinderP AtomNameC ann) n l
+      zipNest Empty [] = Empty
+      zipNest (Nest b bs) (a:anns) = withExtEvidence b $ withSubscopeDistinct bs $
+        Nest (b:>a) $ zipNest bs $ sinkList anns
+      zipNest _ _ = error "Mismatched lengths in NonDepNest"
+
+instance BindsEnv DataConRefBinding where
```

Comment:
Why is this instance necessary if it's not implemented?  Under what circumstances might this error occur at runtime?

Reply:
Well it might occur if you even use this method, but at the moment it is guaranteed to be _dynamically_ (but not _statically_) dead. This is because right now we only use it in the code generated for DCE, but at all stages when DCE happens, the `XYZRef` constructors are not allowed to appear in the IR. It's basically an unfortunate effect of putting all pieces that might ever appear in Dex Core in a single AST.

Reply:
I added a comment.